interface/patient_tracker/patient_tracker.php

```diff
@@ -261,295 +319,421 @@ function openNewTopWindow(newpid,newencounterid) {
+                            continue;
+                        }
+                      // Calendar Appt status gets updated.
+                      // We have added Appt statuses to list apptstatus to include the following categories SMS,AVM,EMAIL,CALL
+
+                        if (($row['msg_reply'] == "CONFIRMED")||($appointment[$row['msg_type']]['stage']=="CONFIRMED")) { //all done then, no need to show any more SMS stuff
+                            $appointment[$row['msg_type']]['stage'] = "CONFIRMED";
+                            $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['CONFIRMED']['html'];
+                         //  $appointment['status'] = $row['msg_type'];
+
+                        } elseif (($row['msg_reply'] == "READ")||($appointment[$row['msg_type']]['stage']=="READ")) {
+                            if ($appointment[$row['msg_type']]['stage']  != "CONFIRMED") {
+                                $appointment['stage'] = "READ";
+                                $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['READ']['html'];
+                            }
+                        } elseif (($row['msg_reply'] == "SENT")||($appointment[$row['msg_type']]['stage']=="SENT")) {
+                            if (($appointment[$row['msg_type']]['stage']!="CONFIRMED")&& ($appointment[$row['msg_type']]['stage']!="READ")) {
+                                $appointment[$row['msg_type']]['stage'] = "SENT";
+                                $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['SENT']['html'];
+                            }
+                        } elseif (($row['msg_reply'] == "To Send")||($appointment['stage']=="QUEUED")) {
+                            if (($appointment[$row['msg_type']]['stage']!="CONFIRMED")&&($appointment[$row['msg_type']]['stage']!="READ")&&($aappointment[$row['msg_type']]['stage']!="SENT")) {
+                                $appointment[$row['msg_type']]['stage']   = "QUEUED";
+                                $icon_here[$row['msg_type']] = $icons[$row['msg_type']]['SCHEDULED']['html'];
+                            }
+                        }
+                      //these are additional icons if present
+                        if (($row['msg_reply'] == "CALL")||($row['msg_type']=="NOTES")) {
+                            if (($appointment['NOTES']['staged'])||($row['msg_type']=="NOTES")) {
+                                $appointment['NOTES']['staged'] ="YES";
+                                $icon_4_CALL = $icons['NOTES']['CALLED']['html'];
+                            } else {
+                                $icon_4_CALL = $icons[$row['msg_type']]['CALL']['html'];
+                            }
+                            $icon_CALL = "<span onclick=\"doCALLback('".$date_squash."','".$appointment['eid']."','".$appointment['pc_cattype']."')\">".$icon_4_CALL."</span>
```

Comment:
we need to do something a bit weird when literals parameters within functions(also note I added text() to $icon_4_CALL, however only do this is there is not html in that variable):
``````
$icon_CALL = "<span onclick=\"doCALLback('".attr(addslashes($date_squash.))"','".attr(addslashes($appointment['eid']))."','".attr(addslashes($appointment['pc_cattype']))."')\">".text($icon_4_CALL)."</span>
``````
(see the last exception in step 2 here for better explanation: http://www.open-emr.org/wiki/index.php/Codebase_Security#SQL-Injection_and_Cross-Scripting_Prevention )